### PR TITLE
caching

### PR DESCRIPTION
**Pull request by elizata7cbs**

### CODE REVIEW

File: FXVault/views.py [hunk 1]
```diff
@@ -14,7 +16,11 @@ class TransactionCreateView(generics.CreateAPIView):
     permission_classes = [IsAuthenticated]
 
     def get(self, request):
-        return Response({"message": "You are authenticated!"})
```
Comment (abedkiloo) [inline FXVault/views.py:17]: The permission call should handle this for you

File: FXVault/views.py [hunk 2]
```diff
@@ -23,60 +29,132 @@ def create(self, request, *args, **kwargs):
...
+                            "status": status.HTTP_400_BAD_REQUEST,
+                            "message": f"Conversion rate for {output_currency} not available."
+                        }, status=status.HTTP_400_BAD_REQUEST)
                 else:
-                    return Response(
-                        {"error": f"Conversion rate for {output_currency} not available."},
-                        status=status.HTTP_400_BAD_REQUEST,
-                    )
+                    return Response({
+                        "data": {},
```
Comment (abedkiloo) [inline FXVault/views.py]: add error:None

File: FXVault/views.py [hunk 3]
```diff
@@ -23,60 +29,132 @@ def create(self, request, *args, **kwargs):
...
+                        }, status=status.HTTP_201_CREATED, headers=headers)
+                    else:
+                        return Response({
+                            "data": {},
+                            "status": status.HTTP_400_BAD_REQUEST,
+                            "message": f"Conversion rate for {output_currency} not available."
+                        }, status=status.HTTP_400_BAD_REQUEST)
                 else:
-                    return Response(
-                        {"error": f"Conversion rate for {output_currency} not available."},
```
Comment (abedkiloo) [inline FXVault/views.py:42]: add { "data": empty dictionary, and status}

File: FXVault/views.py [hunk 4]
```diff
@@ -23,60 +29,132 @@ def create(self, request, *args, **kwargs):
...
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+                # If exchange rate is cached, use it
+                exchange_rate_decimal = Decimal(str(exchange_rate))
+                output_amount = input_amount * exchange_rate_decimal
+                serializer.save(output_amount=output_amount)
+
+                headers = self.get_success_headers(serializer.data)
+                return Response({
+                    "data": serializer.data,
+                    "status": status.HTTP_201_CREATED,
```
Comment (abedkiloo) [inline FXVault/views.py]: Add error : empty value

File: FXVault/views.py (hunk 1 above)
Comment (elizata7cbs) [reply FXVault/views.py:17]: done

File: FXVault/views.py (hunk 2 above)
Comment (elizata7cbs) [reply FXVault/views.py]: done

File: FXVault/views.py (hunk 3 above)
Comment (elizata7cbs) [reply FXVault/views.py:42]: done

File: FXVault/views.py (hunk 4 above)
Comment (elizata7cbs) [reply FXVault/views.py]: handled

File: FXVault/views.py [hunk 5]
```diff
@@ -23,60 +23,136 @@ def create(self, request, *args, **kwargs):
             input_amount = serializer.validated_data.get("input_amount")
```
Comment (abedkiloo) [inline FXVault/views.py]: Use serializer to check whether the data you sent is valid

Comment (elizata7cbs) [reply FXVault/views.py]: used serializer for data validation, updated views, and modified settings

**Pull request merged**